pageserver/src/tenant/timeline.rs

```diff
@@ -3335,6 +3336,130 @@ impl From<anyhow::Error> for CompactionError {
     }
 }
 
+#[serde_as]
+#[derive(serde::Serialize)]
+struct RecordedDuration(#[serde_as(as = "serde_with::DurationMicroSeconds")] Duration);
+
+#[serde_as]
+#[derive(Default, serde::Serialize)]
+#[serde(untagged)]
+enum DurationRecorder {
+    #[default]
+    #[serde(skip)]
+    NotStarted,
+    Recorded(RecordedDuration, #[serde(skip)] tokio::time::Instant),
+}
+
+impl DurationRecorder {
+    pub fn till_now(&self) -> DurationRecorder {
+        match self {
+            DurationRecorder::NotStarted => {
+                panic!("must only call on recorded measurements")
+            }
+            DurationRecorder::Recorded(_, ended) => {
+                let now = tokio::time::Instant::now();
+                DurationRecorder::Recorded(RecordedDuration(now - *ended), now)
```

Comment:
```suggestion
                DurationRecorder::Recorded(RecordedDuration(ended.elapsed()), now)
```

Reply:
with your suggestion, we get the current time twice, potentially a different timestamp than `now`. Rejecting this suggestion, OK?

Reply:
Oh I realize it is also recorded later, then LGTM 🤣